test/interface/optimize.jl

```diff
@@ -0,0 +1,102 @@
+
+using Test
+
+@testset "optimize" begin
+    seed = (0x38bef07cf9cc549d)
+    rng  = StableRNG(seed)
+
+    T = 1000
+    modelstats = normallognormal_meanfield(Float64; rng)
+
+    @unpack model, μ_true, L_true, n_dims, is_meanfield = modelstats
+
+    # Global Test Configurations
+    b⁻¹  = Bijectors.bijector(model) |> inverse
+    q₀_η = TuringDiagMvNormal(zeros(Float64, n_dims), ones(Float64, n_dims))
+    q₀_z = Bijectors.transformed(q₀_η, b⁻¹)
+    obj  = ADVI(10)
+
+    adbackend = AutoForwardDiff()
+    optimizer = Optimisers.Adam(1e-2)
+
+    rng  = StableRNG(seed)
+    q_ref, stats_ref, _ = optimize(
+        rng, model, obj, q₀_z, T;
+        optimizer,
+        show_progress = false,
+        adbackend,
+    )
+    λ_ref, _ = Optimisers.destructure(q_ref)
+
+    @testset "default_rng" begin
+        optimize(
+            model, obj, q₀_z, T;
+            optimizer,
+            show_progress = false,
+            adbackend,
+        )
+
+        λ₀, re  = Optimisers.destructure(q₀_z)
+        optimize(
+            model, obj, re, λ₀, T;
+            optimizer,
+            show_progress = false,
+            adbackend,
+        )
+    end
+
+    @testset "restructure" begin
+        λ₀, re  = Optimisers.destructure(q₀_z)
+
+        rng  = StableRNG(seed)
+        λ, stats, _ = optimize(
+            rng, model, obj, re, λ₀, T;
+            optimizer,
+            show_progress = false,
+            adbackend,
+        )
+        @test λ     == λ_ref
+        @test stats == stats_ref
+    end
+
+    @testset "callback" begin
+        rng  = StableRNG(seed)
+        test_values = rand(rng, T)
+
+        callback(; stat, args...) = begin
+            (test_value = test_values[stat.iteration],)
+        end
```

Comment:
```suggestion
        callback(; stat, args...) = (test_value = test_values[stat.iteration],)
```
The `begin ... end` syntax for functions is not really considered "stylish" anymore:)

Reply:
Trends change so fast...